game/src/engine/audio/AudioService.cpp

```diff
@@ -61,15 +61,17 @@ void AudioService::AddSource(std::string file_name)
     alSourcei(source, AL_LOOPING, AL_FALSE);
     alSourcei(source, AL_BUFFER, buffer);  // GIVE SOURCE ITS BUFFER
 
+    SourceBufferPair source_buffer_pair = {source, buffer};
+
     if (alGetError() != AL_NO_ERROR)
     {
         Log::error("Couldn't create source for {}", file_name);
     }
 
-    non_diegetic_sources_.insert({file_name, {source, buffer}});
+    non_diegetic_sources_.insert({file_name, source_buffer_pair});
 }
 
-void AudioService::AddSource(std::string file_name, std::uint32_t entity_id)
+void AudioService::AddSource(std::uint32_t entity_id, std::string file_name)
```

Comment:
Doesn't really matter tbh, but for the sake of you saving a whole 5 chars when typing in the future, you don't actually need the `std::` prefix for types like `uint32_t`, since they come from the `<cstdint>` header from the C standard libs :D

Reply:
wow, biggest facepalm moment lmao